### ISSUE TITLE
missing function:

### ISSUE THREAD

**Issue by hugoam** [emscripten-core/emscripten#8335]
Since switching from 1.38.18 to 1.38.30, I get a bunch of "missing function" from GL functions that were perfectly fine right before I updated my SDK.

I've tried to clear the cache just to be sure it's not due to some residue from switching.
```
emcc --clear-cache
emcc --clear-ports
```

I don't even understand how those functions can go missing. They are implemented in emscripten webgl and webgl2 libs, that implementation just doesn't seem to carry through the JS output, and I don't understand why. Googling for that "missing function" issue turns up mostly stuff that is a few years old and irrelevant to my issue.

```
  function _emscripten_glUnmapBuffer() {
  err('missing function: emscripten_glUnmapBuffer'); abort(-1);
  }
  function _emscripten_glMapBufferRange() {
  err('missing function: emscripten_glMapBufferRange'); abort(-1);
  }
  function _emscripten_glGetInternalformativ() {
  err('missing function: emscripten_glGetInternalformativ'); abort(-1);
  }
  function _emscripten_glGetBufferPointerv() {
  err('missing function: emscripten_glGetBufferPointerv'); abort(-1);
  }
```

EDIT: Probably related to this one ?
https://github.com/emscripten-core/emscripten/issues/8174
Which is recent enough that it would seem a bug was recently introduced

**Comment by hugoam** [comment]
That's probably not it, but looking a bit at this commit: https://github.com/emscripten-core/emscripten/commit/ef230f55772b7a2b2d4c6a29b8f1cde915bd8e3f (it's the biggest recent change I could find related to this issue): the only weird thing I found is that `glGetBufferPointerv`, `glMapBufferRange`, `glUnmapBuffer`, `glFlushMappedBufferRange` are declared in emscripten_webgl2_get_proc_address while they are actually part of library_webgl.js (and not library_webgl2.js). But that's probably just an inconsistency, I don't see how that might actually cause issues in the compiler pipeline. Also `glGetInternalformativ`, is consistently in webgl2.

**Comment by hugoam** [comment]
Investigating further, about `glGetInternalformativ`, it's very possible that the issue was already there, but that I only notice it now because it was recently added to the get_proc_address. (So before, as it was null, the code never tried calling it). I'll check tomorrow with `1.38.18` to see if it was already implemented as `missing function` on the JS side...

**Comment by hugoam** [comment]
I'm starting to understand a bit better the inter-mingling of different issues that are at play here. Also, for context, this is all in a get_proc_address based codebase/setup. No GL functions are ever linked statically. I built with `1.38.18` again, to see what functions were missing, and there was actually a ton more of them:
- Most of the functions that were `err('missing function: emscripten_xxx')` , were just not implemented by emscripten, and were simply removed from the `get_proc_adress` implementation in the recent refactors, which, I suppose, means they are not linked anymore as they are not used anywhere. This is nice, emitted JS is smaller.
- Some functions were added to the `get_proc_address` implementation: that's the case of `glGetInternalformativ`, `glMapBufferRange`. So they weren't even present in the emitted JS earlier, nor accessible through `get_proc_address`, which is probably why no one ever noticed this issue. But since they have been added, it is linked in, and `get_proc_address` does return a valid address, except it's the adress of a non-implemented JS function. But those functions are indeed implemented in https://github.com/emscripten-core/emscripten/blob/1.38.30/src/library_webgl2.js#L64 , https://github.com/emscripten-core/emscripten/blob/1.38.30/src/library_webgl.js#L3357 so someone more knowledgeable about emscripten will have to try to understand why they are not emitted properly.
- Some of those were already present in `get_proc_address`: that's the case of `glGetBufferPointerv` and `glUnmapBuffer`. I verified, and they were already implemented as `missing function` in `1.38.18`. So the issue is not an entirely new issue either.

**Comment by hugoam** [comment]
I think I finally got to the bottom of it :)
- glGetInternalformativ is due to a typo in how it's spelled in https://github.com/emscripten-core/emscripten/blob/1.38.30/src/library_webgl2.js#L64. I opened a pull request: https://github.com/emscripten-core/emscripten/pull/8337
- the glMapBufferRange issues are related to not activating FULL_ES3, but at least a more precise error message in the output code would be useful. Those functions could be implemented to report that FULL_ES3 needs to be specified at build time in an #else block after https://github.com/emscripten-core/emscripten/blob/1.38.30/src/library_webgl.js#L3320, instead of the generic "missing function" resulting of those functions not being defined at all.
